app/models/piece.rb

```diff
@@ -30,20 +41,40 @@ class Piece < ApplicationRecord
 
 
   def vertical_obstruction?(destination_y)
-    range = self.y_position < destination_y ? (self.y_position..destination_y) : (destination_y..self.y_position)
+    range = self.y_position < destination_y ? (self.y_position...destination_y) : (destination_y...self.y_position)
     game.pieces.where(
-        color: self.color,
-        x_position: self.x_position, 
+        # color: self.color,
```

Comment:
We need to have color because piece can only be obstructed by pieces with the same color so we are finding all the pieces of the same color in between.

Reply:
I was thinking about that as well, but then I remember the Pawn can be obstructed by any piece in front of it since Pawn can only capture pieces diagonally.

Reply:
I guess

Reply:
If you want to be sure, I can definitely uncomment it. I don't think it's wrong, I just think it's not necessary in this case. And I think the tests should still pass if I uncomment it.